src/iarc7_motion/iarc_tasks/height_holder.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+# A helper class for a task that will spit out a velocity that's appropriate
+# to reach and hold a desired position in the z dimension
+# Uses a simple p-controller
+
+import math
+import rospy
+
+from geometry_msgs.msg import TwistStamped
+from nav_msgs.msg import Odometry
+
+class HeightHolder():
+    def __init__(self):
+        self._k_z = .1 #p-term for z-height
+        self._odometry = None
+        self._current_velocity_sub = rospy.Subscriber('/odometry/filtered',
+                                              Odometry,
+                                              self._current_velocity_callback)
+        try:
+            self._TRACK_HEIGHT = rospy.get_param('~track_roomba_height')
+        except KeyError as e:
+            rospy.logerr('Could not lookup a parameter for track roomba task')
+            raise
+
+        while self._odometry is None:
```

Comment:
this is partially addressed with the refactored state checking and I removed the while loop. I think with the new state checking logic this should be handled much better, or at least in a more readable form.